Lecture20(Munchkin)/ManchkinGame_HW/ModifierDeck.cpp

```diff
@@ -0,0 +1,91 @@
+#include "ModifierDeck.h"
+#include "DeckSelectionHelper.h"
+#include "Modifier.h"
+#include <fstream>
+#include <sstream>
+
+ModifierDeck::ModifierDeck() : m_modifiersDatabase(), m_deck(m_modifiersDatabase)
+{
+	std::ifstream file("modifiers_database.txt");
+	if (!file.is_open())
+	{
+		throw std::runtime_error("Failed to open items database file.");
+	}
+
+	std::string line;
+	while (std::getline(file, line))
+	{
+		std::stringstream ss(line);
+		std::string type, paramStr;
+		std::getline(ss, type, ';');
+		std::getline(ss, paramStr);
+		if (!paramStr.empty())
+		{
+			if (type == "SimpleModifier")
+			{
+				int param = std::stoi(paramStr);
+				m_modifiersDatabase.push_back(new SimpleModifier{ param });
+			}
+			else if (type == "HalvesMonsterLevel")
+			{
+				if (paramStr == "Undead")
+				{
+					m_modifiersDatabase.push_back(new HalvesMonsterLevel{ Tribe::Undead });
+				}
+				else if (paramStr == "God")
+				{
+					m_modifiersDatabase.push_back(new HalvesMonsterLevel{ Tribe::God });
+				}
+				else
+				{
+					throw std::runtime_error("Unknown tribe: " + paramStr);
```

Comment:
just make it of default tribe, it's not the choice to crash the game if it's in a way minor data setup misconfiguration. Warning into a log would be more than enough